### PR TITLE
:bug: CI should pass when on codecov upload fail.

### PR DESCRIPTION
**Pull request by xuezhaojun**
## Summary

Codecov becomes because github api outage sometimes. Change to pass CI job when the codecov upload is failed.

## Related issue(s)

https://github.com/codecov/codecov-action/issues/926

### CODE REVIEW

File: .github/workflows/pre.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ jobs:
           flags: unit
           name: unit
           verbose: true
-          fail_ci_if_error: true
+          fail_ci_if_error: false
```
Comment (zhujian7) [inline .github/workflows/pre.yml:72]: add a TODO like change this back when we find a way to stabilize it?

Comment (xuezhaojun) [reply .github/workflows/pre.yml:72]: That's a good point to discuss on, I prefer leave it be to false.  And if codecov continously being unstable we transfer to another tool.

**Pull request merged**